### PR TITLE
[cu-1ubuznb] Vesting Benchmarks

### PR DESCRIPTION
**Pull request by vimukthi-git**
This PR adds benchmark tests to Vesting pallet according to the [guide](https://github.com/ComposableFi/composable/blob/main/docs/benchmarking.md).

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-1ubuznb Vesting pallet benchmarking](https://app.clickup.com/t/1ubuznb)

**Comment by PoisonPhang** [comment]
You'll want to format with nightly to pass linter checks
`cargo +nightly fmt`

**Comment by hussein-aitlahcen** [comment]
> You'll want to format with nightly to pass linter checks `cargo +nightly fmt`

We have the `make style` command for this :)

**Comment by hussein-aitlahcen** [comment]
We probably need to add this to the `dali` runtime as well as the vesting pallet is already used here :)

**Comment by vimukthi-git** [comment]
@hussein-aitlahcen Sure I will try that. Right now I'm trying to fix an incompatibility between the mock `Runtime` and the benchmark tests for `VestedTransferOrigin` type.

File: frame/vesting/src/lib.rs [hunk 1]
```diff
@@ -56,6 +56,8 @@ use sp_runtime::{
 };
 use sp_std::{convert::TryInto, vec::Vec};
 
+#[cfg(feature = "runtime-benchmarks")]
```
Comment (dzmitry-lahoda) [inline frame/vesting/src/lib.rs:59]: i would prefer bench run as tests too

File: frame/vesting/src/mock.rs [hunk 2]
```diff
@@ -75,21 +78,33 @@ impl frame_system::Config for Runtime {
 	type MaxConsumers = frame_support::traits::ConstU32<16>;
 }
 
+fn benchmark_vested_transfer_acc() -> AccountId {
```
Comment (dzmitry-lahoda) [inline frame/vesting/src/mock.rs:81]: i would prefer _account

**Comment by hussein-aitlahcen** [comment]
@vimukthi-git, I've added the label `needs-benchmarks` to regenerate the benchmarks from the canonical CI machine, we probably need to rebase to keep the new file only and skip the other so that you don't have to deal with conflicts.

**Review by hussein-aitlahcen — APPROVED** [review]

**Review by dzmitry-lahoda — DISMISSED** [review]

**Review by dzmitry-lahoda — APPROVED** [review]

**Review by hussein-aitlahcen — APPROVED** [review]

**Pull request merged**